tag.go

```diff
@@ -108,11 +125,13 @@ func ParseQualifiedTagName(qtn string) ([]string, error) {
 				if closing == -1 {
 					return nil, fmt.Errorf("Field #%d: '[' without ']'", i+1)
 				}
-				idx := a[:closing]
-				if _, err := strconv.ParseUint(idx, 10, 32); err != nil {
-					return nil, fmt.Errorf("Field #%d: Invalid array index: %v", i+1, err)
+				arrayContents := a[:closing]
+				for _, arrayIdx := range strings.Split(arrayContents, ",") {
+					if _, err := strconv.ParseUint(arrayIdx, 10, 32); err != nil {
+						return nil, fmt.Errorf("Field #%d: Invalid array index %v: %v", i+1, arrayIdx, err)
```

Comment:
`arrayIdx` isn't an index anymore. It's a string. Slightly confusing.

Reply:
I believe this test case should pass:
```go
	{"ARRAY[ 0 ,  1  , 2 ]", []string{"ARRAY", "0", "1", "2"}, ""},
```
I think `libplctag` has examples of spaces like that. Or at least handles that in its code. (I'm not sure if `libplctag` has an incorrect grammar spec or incorrect implementation.)

...and then I went and tried to make the change myself, which I probably shouldn't have done. (I thought it would be a one-line addition of `strings.TrimSpace`, but of course that didn't work because you reject incoming strings with spaces anywhere.) It's on a branch (same name, but at `james/*`).
I think my branch works, but it changes a lot of things (and might need a bit of cleanup). Maybe it's not worth adding if we just avoid spaces in multi-arrays. Doesn't matter to me.

Reply:
I'm happy to use your patch.  If we need to touch this code again maybe we can revisit a refactoring, but I doubt we'd ever have to touch it again so it's probably fine if it's slightly gnarly.